File: RxTools-library/src/main/java/com/vondear/rxtools/view/wheelhorizontal/NumericWheelAdapter.java
/*
 * android-spinnerwheel
 * https://github.com/ai212983/android-spinnerwheel
 *
 * based on
 *
 * Android Wheel Control.
 * https://code.google.com/p/android-wheel/
 *
 * Copyright 2011 Yuri Kanivets
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.vondear.rxtools.view.wheelhorizontal;

import android.content.Context;

/**
 * Numeric Wheel adapter.
 */
public class NumericWheelAdapter extends AbstractWheelTextAdapter {
    
    /** The default min value */
    public static final int DEFAULT_MAX_VALUE = 9;

    /** The default max value */
    private static final int DEFAULT_MIN_VALUE = 0;
    
    // Values
    private int minValue;
    private int maxValue;
    
    // format
    private String format;
    
    /**
     * Constructor
     * @param context the current context
     */
    public NumericWheelAdapter(Context context) {
        this(context, DEFAULT_MIN_VALUE, DEFAULT_MAX_VALUE);
    }

    /**
     * Constructor
     * @param context the current context
     * @param minValue the spinnerwheel min value
     * @param maxValue the spinnerwheel max value
     */
    public NumericWheelAdapter(Context context, int minValue, int maxValue) {
        this(context, minValue, maxValue, null);
    }

    /**
     * Constructor
     * @param context the current context
     * @param minValue the spinnerwheel min value
     * @param maxValue the spinnerwheel max value
     * @param format the format string
     */
    public NumericWheelAdapter(Context context, int minValue, int maxValue, String format) {
        super(context);
        
        this.minValue = minValue;
        this.maxValue = maxValue;
        this.format = format;
    }

    @Override
    public CharSequence getItemText(int index) {
        if (index >= 0 && index < getItemsCount()) {
            int value = minValue + index;
            return format != null ? String.format(format, value) : Integer.toString(value);
        }
        return null;
    }

    @Override
    public int getItemsCount() {
        return maxValue - minValue + 1;
    }    
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/dialog/wheel/NumericWheelAdapter.kt
package com.tamsiree.rxui.view.dialog.wheel

import android.content.Context

/**
 * @author tamsiree
 * Numeric Wheel adapter.
 */
open class NumericWheelAdapter

/**
 * Constructor
 * @param context the current context
 */
@JvmOverloads
constructor(context: Context, // Values
            private val minValue: Int = DEFAULT_MIN_VALUE, private val maxValue: Int = DEFAULT_MAX_VALUE, // format
            private val format: String? = null) : AbstractWheelTextAdapter(context) {

    public override fun getItemText(index: Int): CharSequence? {
        if (index in 0 until itemsCount) {
            val value = minValue + index
            return if (format != null) String.format(format, value) else value.toString()
        }
        return null
    }

    override val itemsCount: Int
        get() = maxValue - minValue + 1

    companion object {
        /** The default min value  */
        const val DEFAULT_MAX_VALUE = 9

        /** The default max value  */
        private const val DEFAULT_MIN_VALUE = 0
    }

}